src/openms/source/FORMAT/MzTabFile.cpp

```diff
@@ -2958,14 +2970,17 @@ namespace OpenMS
     {
       MzTabPSMSectionRow row;
       bool first = true;
+      size_t n_header_columns = 0;
       while (s.nextPSMRow(row))
       {
         if (first)
         { // add header
-          tab_file << "\n" << generateMzTabPSMHeader_(n_search_engine_scores, s.getPSMOptionalColumnNames()) + "\n";
+          tab_file << "\n" << generateMzTabPSMHeader_(n_search_engine_scores, s.getPSMOptionalColumnNames(), n_header_columns) + "\n";
           first = false;
         }
-        tab_file << generateMzTabSectionRow_(row, s.getPSMOptionalColumnNames(), meta_data) + "\n";
+        size_t n_section_columns = 0;
+        tab_file << generateMzTabSectionRow_(row, s.getPSMOptionalColumnNames(), meta_data, n_section_columns) + "\n";
+        if (n_header_columns != n_section_columns)  throw Exception::Postcondition(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "PSM header and content differs in columns. Please report this bug to the OpenMS developers.");
```

Comment:
now makes sure that the number of header columns never differ from the number of section columns

Reply:
as a side effect: also makes sure that all section rows have the same number of columns